docs/vector-library.md

```diff
@@ -0,0 +1,111 @@
+# Vector library
+
+## Summary
+
+Implement a standard library that provides functionality for the vector type.
+
+## Motivation
+
+Currently, vectors are a primitive type implemented internally. All of the heavy work to implement vectors is done, but there is no runtime-agnostic way to use vectors in Luau code. Individual Luau runtimes, such as [Lune](https://github.com/lune-org/lune) or Roblox, must provide their own distinct libraries to work with native vectors which may be inconsistent. In cross-runtime code, this results in performance drawbacks & difficulty utilizing native vectors.
+
+## Design
+
+The default metatable for vectors should now be the `vector` library. While buffers and coroutines do not have this, vectors likely should for ergonomics.
+
+### Library functions & constants
+
+This RFC proposes the following basic functions & constants as a starting point, but as with all builtin libraries, future RFCs can propose additional functions.
+
+---
+
+`vector(x: number?, y: number?, z: number?)`
```

Comment:
```suggestion
`vector(x: number, y: number, z: number)`
```

It's best to encourage practice of hitting the fastest constructor we have available.

It will also avoid confusion of what vector(1) is.
While `Vector3.new(1)` is [1, 0, 0] on Roblox, for people coming from GLSL where `vec3(1)` is [1, 1, 1], it might be unintuitive.
Even if supported by runtime, the typing can be more strictly defined.